dashboard-all-the-things/src/app/dashboards/stackoverflow/stackoverflow.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { Question } from './Question';
+import { WidgetEvent } from './../WidgetEvent';
+import { DashboardComponent } from './../DashboardComponent';
+
+@Component({
+  selector: 'app-stackoverflow',
+  templateUrl: 'stackoverflow.component.html',
+  styleUrls: ['stackoverflow.component.scss']
+})
+export class StackOverflowComponent implements DashboardComponent {
+
+  private questions: Array<Question>;
+  private questionToShow: Question;
+
+  public update(event: WidgetEvent) {
+    this.questions = event.payload.questions;
+    this.questionToShow = this.pickRandomQuestion();
+  }
+
+  private pickRandomQuestion(): Question {
+    let question = this.questions[Math.floor(Math.random() * (this.questions.length))];
```

Comment:
Since this logic is duplicated maybe you want to create a Service that extracts a random Object from an Array?